src/utils/init/utils.js

```diff
@@ -0,0 +1,136 @@
+const path = require('path')
+
+const chalk = require('chalk')
+const cleanDeep = require('clean-deep')
+const dotProp = require('dot-prop')
+const inquirer = require('inquirer')
+const isEmpty = require('lodash/isEmpty')
+
+const { fileExistsAsync, writeFileAsync } = require('../../lib/fs')
+
+const normalizeDir = ({ siteRoot, dir, defaultValue }) => {
+  if (dir === undefined) {
+    return defaultValue
+  }
+
+  const relativeDir = path.relative(siteRoot, dir)
+  return relativeDir || defaultValue
+}
```

Comment:
I am a little confused about what this function does.

`relativeDir` can also be falsy if an empty string is returned by `path.relative()`, meaning `path.normalize(siteRoot) === path.normalize(dir)`. I believe `siteRoot` and `dir` are already `path.normalize()`'d, so that would mean `siteRoot === dir`. 

This would mean the `defaultValue` is assigned if `dir` is either `undefined` or equal `siteRoot`. Am I misunderstanding it?

Reply:
> This would mean the `defaultValue` is assigned if `dir` is either `undefined` or equal `siteRoot`. Am I misunderstanding it?

Yes `defaultValue` is assigned if `dir` is either `undefined` or equal to `siteRoot`.
The latter case happens when `build.publish === '.'` so we would like to use `.` as a default value and not an empty string

Reply:
:+1: